src/sentry/shared_integrations/client/base.py

```diff
@@ -248,16 +293,23 @@ def _request(
                 # Rather than worrying about what the other layers might be, we just stringify to detect this.
                 if "ConnectionResetError" in str(e):
                     self.track_response_data("connection_reset_error", span, e)
+                    self.record_error(e)
                     raise ApiConnectionResetError("Connection reset by peer", url=full_url) from e
                 # The same thing can happen with an InvalidChunkLength exception, which is a subclass of HTTPError
                 if "InvalidChunkLength" in str(e):
                     self.track_response_data("invalid_chunk_length", span, e)
+                    self.record_error(e)
                     raise ApiError("Connection broken: invalid chunk length", url=full_url) from e
 
                 # If it's not something we recognize, let the caller deal with it
                 raise e
 
             self.track_response_data(resp.status_code, span, None, resp)
+            self.record_response(
```

Comment:
Hmm this seems to be causing [this test failure](https://github.com/getsentry/sentry/actions/runs/5616794974/job/15219775578?pr=52994) (search for "failures", annoying it doesn't automatically go to it) - is there a reason not to do it like:

`self.record_response(resp)`? 

You'd have to update the Slack client (`src/sentry/integrations/slack/client.py`
) too like:
```
    def is_response_fatal(self, response: Response) -> bool:
        resp_json = response.json()
        if not resp_json.get("ok"):
            if "account_inactive" == resp_json.get("error", ""):
                return True
```

Reply:
When I used self.record_response(resp) that caused this error:
`AttributeError: 'function' object has no attribute 'get'` in `is_response_fatal` in `src/sentry/integrations/slack/client.py` 
so I think I need the full `BaseApiResponse` since I get the same errors with the changes you suggested. I'm also confused why line 319 does not cause the test failures from `BaseApiResponse.from_response()`